File: Data Structures & Algorithms/maximum-subarray/submission-7.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        curr = nums[0]
        maxsum = nums[0]

        for i in range(1,len(nums)):
            if curr + nums[i] > nums[i]:
                curr = curr + nums[i]
            else:
                curr = nums[i]
            maxsum = max(maxsum, curr)
        
        return maxsum

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-7.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> seen;
        for(int x: nums){
            seen.insert(x);
        }

        int max_it = 0;
        for(int x: nums){
            if(seen.find(x-1) != seen.end()){
                continue;
            }

            int cnt = 1;
            int current = x;
            while(seen.find(current + 1) != seen.end()){
                current = current + 1;
                cnt++;
            }
            max_it = max(max_it, cnt);
        }
        return max_it;
    }
};
